File: back/app/models/vehiculo.py
from app.extensions import db
from app.models.base import BaseModel


class Vehiculo(BaseModel):
  __tablename__ = 'vehiculos'

  marca = db.Column(db.String(80), nullable=False)
  modelo = db.Column(db.String(80), nullable=False)
  version = db.Column(db.String(80))
  anio = db.Column(db.Integer)
  patente = db.Column(db.String(20), unique=True)
  estado = db.Column(db.String(40), default='disponible')
  cliente_id = db.Column(db.Integer, db.ForeignKey('clientes.id'))

  def __repr__(self):
    return f'<Vehiculo {self.marca} {self.modelo}>'

File: desktop/src/PipitaDesktop/Models/Vehiculo.cs
namespace PipitaDesktop.Models;

public sealed class Vehiculo : BaseEntity
{
    public string Marca { get; set; } = string.Empty;
    public string Modelo { get; set; } = string.Empty;
    public string? Version { get; set; }
    public int? Anio { get; set; }
    public string? Patente { get; set; }
    public string Estado { get; set; } = "disponible";

    public int? ClienteId { get; set; }
    public Cliente? Cliente { get; set; }

    public ICollection<Servicio> Servicios { get; set; } = new List<Servicio>();
    public ICollection<SolicitudCliente> Solicitudes { get; set; } = new List<SolicitudCliente>();
    public ICollection<TrabajoDistribuidora> TrabajosDistribuidora { get; set; } = new List<TrabajoDistribuidora>();
    public ICollection<Cita> Citas { get; set; } = new List<Cita>();
}


